src/swap/types.ts

```diff
@@ -1,23 +1,68 @@
 import { Address } from 'viem';
-import { Token } from '../token/types';
+import { LegacyTokenData, Token } from '../token/types';
 
 export type AddressOrETH = Address | 'ETH';
 
+/**
+ * Note: exported as public Type
+ */
+export type GetQuoteParams = {
```

Comment:
For all types, let's add a comment that explain each line, similar to what we did here https://github.com/coinbase/onchainkit/blob/main/src/token/types.ts#L50

And keep all types alphabetical ordered.